ci/rust-build-main.bat

```diff
@@ -30,6 +30,7 @@ pushd rust
 rustup default nightly
 rustup show
 cargo build --target %TARGET% --all-targets --release || exit /B
+cargo build --target %TARGET% --all-targets --no-default-features --release || exit /B
```

Comment:
Given that in one of the earlier PRs we reduced the amount that this script does (https://github.com/apache/arrow/commit/83a4e979271535b74de9870289cf99d02f6eb16b#diff-09e63d5097178234f5b3b9d607d9c88aL47), won't we run the risk of increasing build times too much if we build 2 `--release` binaries here?

Reply:
Yea, maybe.  I posted this before the recent issues with build times.  I don't think that this will be used often but we should make sure to test that it at least builds.  I'll change it to a debug build.